src/shared/utils/useAxisTicksGenerator.ts

```diff
@@ -0,0 +1,31 @@
+import {useMemo} from 'react'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+export const useAxisTicksGenerator = ({
+  xTotalTicks,
+  xTickStart,
+  xTickStep,
+  yTotalTicks,
+  yTickStart,
+  yTickStep,
+}) =>
+  useMemo(() => {
+    if (!isFlagEnabled('axisTicksGenerator')) {
```

Comment:
we have to be very careful about how this is deployed if this is done through config cat. the way `isFlagEnabled` is written, you can't be sure that the results will be correct, depending on when the function is called in request's lifecycle